chain/chain/src/chain.rs

```diff
@@ -3307,23 +3307,30 @@ pub fn check_refcount_map(chain: &mut Chain) -> Result<(), Error> {
     let mut chain_store_update = ChainStoreUpdate::new(chain.mut_store());
     for (block_hash, refcount) in block_refcounts {
         match chain_store_update.get_block(&block_hash) {
-            Ok(_) => {
-                if chain_store_update.get_block_refcount(&block_hash)? != &refcount {
+            Ok(_) => match chain_store_update.get_block_refcount(&block_hash) {
+                Ok(&found_refcount) => {
+                    if found_refcount != refcount {
+                        return Err(ErrorKind::Other(format!(
+                            "invalid number of references in Block {:?}, expected {:?}, found {:?}",
+                            block_hash, refcount, found_refcount
+                        ))
+                        .into());
+                    }
+                }
+                Err(e) => {
                     return Err(ErrorKind::Other(format!(
-                        "invalid number of references in Block {:?}, expected {:?}, found {:?}",
-                        block_hash,
-                        chain_store_update.get_block_refcount(&block_hash).unwrap(),
-                        refcount
+                        "Block {:?} is deleted while expected {:?} references; get_block_refcount failed: {:?}",
+                        block_hash, refcount, e
                     ))
                     .into());
                 }
-            }
-            Err(_) => {
+            },
+            Err(e) => {
                 if chain_store_update.get_block_refcount(&block_hash).is_ok() {
                     return Err(ErrorKind::Other(format!(
-                        "Block {:?} expected to be deleted, found {:?} references instead",
+                        "Block {:?} expected to be deleted, found {:?} references instead: get_block failed: {:?}",
                         block_hash,
-                        chain_store_update.get_block_refcount(&block_hash).unwrap(),
+                        chain_store_update.get_block_refcount(&block_hash).unwrap(), e
```

Comment:
Instead of `if value.is_ok() { ... value.unwrap() }` it is better to use:

```rust
if let Ok(block_refcount) = chain_store_update.get_block_refcount(&block_hash) {
   // ...
   block_refcount
   // ...
}
```

Reply:
++